modules/core/src/test/java/org/apache/ignite/internal/processors/security/sandbox/IgnitionComponentProxyTest.java

```diff
@@ -0,0 +1,193 @@
+    @Test
+    public void testIgniteWithName() {
+        Supplier<Ignite> s = new Supplier<Ignite>() {
+            @Override public Ignite get() {
+                return Ignition.ignite(SRV);
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testIgniteWithID() {
+        final UUID srvId = grid(SRV).localNode().id();
+
+        Supplier<Ignite> s = new Supplier<Ignite>() {
+            @Override public Ignite get() {
+                return Ignition.ignite(srvId);
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testAllGrids() {
+        Supplier<Collection<Ignite>> s = new Supplier<Collection<Ignite>>() {
+            @Override public Collection<Ignite> get() {
+                return Ignition.allGrids();
+            }
+        };
+
+        checkCollection(s);
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testStart() {
+        Supplier<Ignite> s = new Supplier<Ignite>() {
+            @Override public Ignite get() {
+                try {
+                    Permissions perms = new Permissions();
+
+                    perms.add(new AllPermission());
+
+                    AccessControlContext acc = AccessController.doPrivileged(
+                        (PrivilegedAction<AccessControlContext>)() -> new AccessControlContext(AccessController.getContext(),
+                            new IgniteDomainCombiner(perms)));
+
+                    return AccessController.doPrivileged((PrivilegedExceptionAction<Ignite>)
+                            () -> Ignition.start(
+                                getConfiguration("node_" + G.allGrids().size())
+                                    .setConnectorConfiguration(null)
+                            ), acc);
+                }
+                catch (Exception e) {
+                    throw new IgniteException(e);
+                }
+            }
+        };
+
+        check(s);
+    }
+
+    /**
+     * @param s Supplier that returns an instance of Ignite.
+     */
+    private void check(final Supplier<Ignite> s) {
+        Supplier<Collection<Ignite>> supplier = new Supplier<Collection<Ignite>>() {
+            @Override public Collection<Ignite> get() {
+                return Collections.singleton(s.get());
+            }
+        };
+
+        checkCollection(supplier);
+    }
+
+    /**
+     * @param s Supplier that returns a collection of Ignite instances.
+     */
+    private void checkCollection(Supplier<Collection<Ignite>> s) {
+        Ignite srv = grid(SRV);
+
+        Ignite clnt = grid(CLNT);
+
+        //Checks that inside the sandbox we should get a proxied instance of Ignite.
+        clnt.compute(clnt.cluster().forNodeId(srv.cluster().localNode().id()))
+            .broadcast(() -> {
+                Collection<Ignite> nodes = s.get();
+
+                for (Ignite node : nodes)
+                    node.compute().call(() -> true);
```

Comment:
It's better to add a comment, why `() -> true` shouldn't work if we get not a proxy instance here.